Develop

TokyoSU opened the pull request:
Don't included GetCollisionInfo() since @asasas9500 decompiled it in this PR #27 
Also dont include CalculateObjectLighting(), it's included in this PR #27

TokyoSU committed:
Decompiled FindGridShift()

TokyoSU committed:
Decompiled ShiftItem()

- Fixed FindGridShift() Code Style.

TokyoSU committed:
Decompiled UpdateLaraRoom()

- Added phd_math to have the same header file as @asasas9500

TokyoSU committed:
Decompiled GetTiltType()

- Fixed FLOOR_INFO structure:
pitRoom and skyRoom where not unsigned !
- Fixed another code style for FindGridShift()

TokyoSU committed:
Fixed GetTiltType()

- This function used short instead of __int16

TokyoSU committed:
Decompiled LaraBaddieCollision()

TokyoSU committed:
Fixed Tab

For unknown reason CodeBlocks don't enable the tab character by default :x

TokyoSU committed:
Decompiled EffectSpaz()

- IDAPro is shit with structure then pointer, Lara.spaz_effect where Lara.last_pos.z + 2

TokyoSU committed:
Decompiled CreatureCollision()

- Added some define function.
- Decompiled ObjectCollision() too.

TokyoSU committed:
Decompiled DoorCollision()

TokyoSU committed:
Decompiled TrapCollision()

TokyoSU committed:
Decompiled ItemPushLara() and TestBoundsCollide()

TokyoSU committed:
Decompiled TestLaraPosition()

TokyoSU committed:
Decompiled AlignLaraPosition()

TokyoSU committed:
Decompiled MoveLaraPosition()

TokyoSU committed:
Decompiled Move3DPosTo3DPos()

Finished collide.cpp.
PS: GetCollisionInfo() was already decompiled by @asasas9500 on PR #27

TokyoSU committed:
Removing some empty line (collide)

- Removing empty line between function in both .h and .cpp.
- Fixed a short to __int16.

TokyoSU committed:
Fixed missing tab

Normally it was the last change.

TokyoSU committed:
Fixed another missing tab

It was when i started and i found out about CodeBlocks not replacing space by tab

TokyoSU committed:
Update TR2_progress.txt

asasas9500 commented on TR2_progress.txt:
```diff
@@ -194,21 +194,21 @@ x function is unused / included in another function
 0x00412FC0:	+	CollideStaticObjects
```
FindGridShift has no marker

TokyoSU replied:
fixed

asasas9500 commented on game/collide.cpp:
```diff
@@ -21,9 +21,30 @@
 #include "global/vars.h"
 
+int __cdecl FindGridShift(int src, int dest) {
+	int srcShift, destShift;
```
One empty line

asasas9500 commented on game/collide.cpp:
```diff
@@ -21,9 +21,30 @@
 
 #include "global/precompiled.h"
 #include "game/collide.h"
+#include "3dsystem/phd_math.h"
```
The order should be the opposite:

```
#include "3dsystem/3d_gen.h"
#include "3dsystem/phd_math.h"

```

TokyoSU replied:
fixed

asasas9500 commented on game/collide.cpp:
```diff
@@ -191,30 +212,463 @@ void __cdecl GetNewRoom(int x, int y, int z, __int16 roomID) {
+}
+
+void __cdecl UpdateLaraRoom(ITEM_INFO* item, int height) {
+	FLOOR_INFO* floor;
```
Pointers on the right side

TokyoSU replied:
fixed

asasas9500 commented on game/collide.cpp:
```diff
@@ -191,30 +212,463 @@ void __cdecl GetNewRoom(int x, int y, int z, __int16 roomID) {
 	DrawRoomsArray[DrawRoomsCount++] = roomID;
 }
 
+void __cdecl ShiftItem(ITEM_INFO* item, COLL_INFO* coll) {
```
Pointers on the right side:

`void __cdecl ShiftItem(ITEM_INFO *item, COLL_INFO *coll) {`

TokyoSU replied:
fixed

asasas9500 commented on game/collide.cpp:
```diff
@@ -191,30 +212,463 @@ void __cdecl GetNewRoom(int x, int y, int z, __int16 roomID) {
+void __cdecl UpdateLaraRoom(ITEM_INFO* item, int height) {
+	FLOOR_INFO* floor;
+	int x, y, z;
+	short roomID;
```
`short` should be `__int16`

TokyoSU replied:
fixed

asasas9500 commented on game/collide.cpp:
```diff
@@ -191,30 +212,463 @@ void __cdecl GetNewRoom(int x, int y, int z, __int16 roomID) {
+BOOL __cdecl TestLaraPosition(__int16* bounds, ITEM_INFO* item, ITEM_INFO* laraitem) {
+	int x, y, z;
+	int xBound, yBound, zBound;
+	short yRot, xRot, zRot;
```
`short` should be `__int16`

TokyoSU replied:
fixed

asasas9500 commented on game/collide.cpp:
```diff
@@ -191,30 +212,463 @@ void __cdecl GetNewRoom(int x, int y, int z, __int16 roomID) {
+
+__int16 __cdecl GetTiltType(FLOOR_INFO* floor, int x, int y, int z) {
+	__int16* data;
+	unsigned char i;
```
`unsigned char` should be `BYTE`

TokyoSU replied:
fixed

asasas9500 commented on game/collide.cpp:
```diff
@@ -191,30 +212,463 @@ void __cdecl GetNewRoom(int x, int y, int z, __int16 roomID) {
+}
+
+void __cdecl EffectSpaz(ITEM_INFO* laraitem, COLL_INFO* coll) {
+	Lara.hit_direction = (unsigned __int16)(laraitem->pos.rotY + PHD_180 - phd_atan(Lara.spaz_effect->pos.z - laraitem->pos.z, Lara.spaz_effect->pos.x - laraitem->pos.x) + PHD_90) >> W2V_SHIFT;
```
`unsigned __int16` should be `UINT16`

TokyoSU replied:
fixed

asasas9500 commented on game/collide.cpp:
```diff
@@ -191,30 +212,463 @@ void __cdecl GetNewRoom(int x, int y, int z, __int16 roomID) {
+		rz -= (bndMax * c - bndMin * s) >> W2V_SHIFT;
+
+		if (spazon && bounds[3] - bounds[2] > 256) {
+			Lara.hit_direction = (unsigned __int16)(laraitem->pos.rotY + PHD_180 - phd_atan(rz, rx) + PHD_90) >> W2V_SHIFT;
```
`unsigned __int16` should be `UINT16`

TokyoSU replied:
fixed

asasas9500 commented on global/types.h:
```diff
@@ -1835,9 +1835,9 @@ typedef struct DoorInfos_t {
 	__int16 index;
 	__int16 box;
-	char pitRoom;
+	unsigned char pitRoom;
```
`unsigned char` should be `BYTE`

TokyoSU replied:
fixed

asasas9500 commented on global/types.h:
```diff
@@ -1835,9 +1835,9 @@ typedef struct DoorInfos_t {
+	unsigned char pitRoom;
 	char floor;
-	char skyRoom;
+	unsigned char skyRoom;
```
`unsigned char` should be `BYTE`

TokyoSU replied:
fixed

asasas9500 commented on game/collide.cpp:
```diff
@@ -191,30 +212,463 @@ void __cdecl GetNewRoom(int x, int y, int z, __int16 roomID) {
+	item = &Items[itemID];
+	if (TestBoundsCollide(item, laraitem, coll->radius)) {
+		if (TestCollision(item, laraitem)) {
+			if (CHK_ANY(coll->flags, 0x8) && Lara.water_status != 1) { // NOTE: original checked "(Lara.water_status == 0) != 2" but it's always true !
```
`Lara.water_status != LWS_Underwater`

TokyoSU replied:
fixed

TokyoSU committed:
Fixed some review from @asasas9500

- Fixed wrong variable type.
- Fixed left pointer to right pointer.
- Fixed missing "+" in TR2_progress.txt
- Fixed missing LWS_Underwater in CreatureCollision()

TokyoSU committed:
Maybe fixed a bug about key

Link: https://discord.com/channels/218414493355606016/394838132643725314/954130260650229790

TokyoSU committed:
Fix MoveLaraPosition() for Flare

TokyoSU committed:
Decompiled DrawLara()

TokyoSU committed:
Decompiled DrawPhaseCinematic() and DrawPhaseGame()

TokyoSU committed:
Decompiled InitInterpolate()

- Added 4 variable (2 int, 1 ptr, 1 array)

TokyoSU committed:
Decompiled phd_PopMatrix_I()

Decompiled phd_PushMatrix_I() too.

TokyoSU committed:
Decompiled phd_RotY_I()

Also decompiled phd_RotX_I() and phd_RotZ_I()

TokyoSU committed:
Decompiled phd_TranslateRel_I()

TokyoSU committed:
Decompiled phd_TranslateRel_ID()

Also decompiled phd_RotYXZ_I()

TokyoSU committed:
Decompiled phd_RotYXZsuperpack_I()

TokyoSU committed:
Fixed DrawLara()

- a phd_PopMatrix() caused weird render artifact.

TokyoSU committed:
Decompiled InterpolateMatrix()

TokyoSU committed:
Decompiled InterpolateArmMatrix()

TokyoSU committed:
Decompiled GetFrames()

TokyoSU committed:
Decompiled GetBoundsAccurate()

TokyoSU committed:
Decompiled GetBestFrame()

- Fixed code style for pointer.

TokyoSU committed:
Update TR2_progress.txt

TokyoSU committed:
Merge branch 'draw' into develop

TokyoSU committed:
Decompiled LaraInitialiseMeshes()

TokyoSU committed:
Decompiled InitialiseLara()

- Fixed warning about ARRAY_SIZE() returning UINT32 instead of INT.
- Used pointer for Lara instead, just because these two variable have the same text size and height (lara & item)

TokyoSU committed:
Decompiled AnimateLara()

- Fixed pointer wrong pos.
- Added EffectFunctions array.
- Removed space between INJECT and in the .H too

TokyoSU committed:
Decompiled LaraUnderWater()

- Added the door cheat from Tomb1Main.
Also:
- Decompiled ShutThatDoor() OpenThatDoor() DoorControl()
- Added 2 new structure.
- Added lara function array !

TokyoSU committed:
Fix space to tab

TokyoSU committed:
Changed short to __int16

Also fixed a warning about a bool always true because of this: soundType != 0x8000 || (waterSurfDist >= 0 || waterSurfDist == NO_HEIGHT)

TokyoSU committed:
Fix lara death in offshore rig

Lara die directly because there is missing variable in InitialiseLara() caused by me :x

TokyoSU commented:
There is bug, need to fix it before pushing !
It's not 100% accurate.

TokyoSU committed:
Fixed LaraBaddieCollision()

TokyoSU committed:
Add array size check

In LaraBaddieCollision()

TokyoSU committed:
Fixed lara not playing some sound

caused by AnimateLara() command pointer being advanced when it's not !

TokyoSU committed:
Fixed a mistake about lara command

TokyoSU committed:
Fixed M16 torso rotation.

TokyoSU committed:
Decompiled GeneralControl()

- Add the bridgelightfix in TR2Main.json to fix the dynamic light created by drawbridge when triggered.

TokyoSU committed:
Bonus level experimental feature

- Crash when trying to load the bonus level.

TokyoSU committed:
Revert "Bonus level experimental feature"

This reverts commit f47ec07bf9af5a8969ec22775ffa28b94e7a170a.

TokyoSU committed:
Fixed a crash

- Decompiled GetJointAbsPosition() and BaddieBiteEffect().
- Changed COLL_INFO flags.
- Fixed lara picking item when underwater, it crashed the game.
- Updated TR2_progress.txt

TokyoSU committed:
Started decompiling setup.cpp

TokyoSU committed:
Finished TrapObjects()

- Fixed my shit about InitialiseFinalLevel() using short argument when there is none.

TokyoSU committed:
Added missing SPIKE_WALL obj

Reworked the TrapObjects() layout, apparently obj->loaded can cause problem if its not a baddy object !

TokyoSU committed:
Finished decompiling ObjectObjects()

- Added SetPickup() to create pickup easily !
- Fixed secret pickup 1 & 3 not having the control behaviour !

TokyoSU committed:
Fixed cinematic crash for level 1

- it was caused by InitialiseGameFlags() not being called anymore since it was missing, it cause crash if a cutscene was played after a level finish.
- Also fixed the hair flotting in air if the cheat was activated when triggering a cutscene !

TokyoSU committed:
Finished setup.cpp

- Decompiled GetCarriedItems()

TokyoSU committed:
Fixed Joystick in DirectInput8

- Wrong flag was used when enumerating joystick.
- Fixed also the vibrating joystick check, it was using double left check instead of left and right check.
- Also fixed using WINBOOL instead of BOOL
- Added all missing define in control.h

TokyoSU committed:
Update TR2_progress.txt

TokyoSU committed:
Decompiled InitialiseCreature()

- if cheat is enabled, when loading a level, you will get all weapons (only for debug dll !)
also added mediapack (small&large) to the bonus game.

TokyoSU committed:
Decompiled CreatureActive()

TokyoSU committed:
Decompiled CreatureAIInfo()

TokyoSU committed:
Increased the count of creature to 16

- Removed InitialiseCreature(), CreatureActive(), CreatureAIInfo().
- Decompiled InitialiseLOTarray(), DisableBaddieAI(), EnableBaddieAI(), InitialiseSlot(), LaraGetNewTarget().
- Now you can have 16 creatures simultaneously in the level.
- Wolf in TR2_GOLD could probably be stuck on wall (trying to climb) or fall 4 click (1 block), it's fixed now.
